### ISSUE TITLE
Limit register email to domain or regex

### ISSUE THREAD

**Issue by HipyCas** [answerdev/answer#167]
**Is your feature request related to a problem? Please describe.**
I am planning on setting up an instance of answer in my uni, but for that I would need to limit the registrations to emails of my university. Creating the users manually is not feasible, as there are too many people to do that by hand and I would have to do so every year.

**Describe the solution you'd like**
I'd like to configure an email domain, like e.g. `mail.myuni.edu`, so you could only register with a mail of the form `xxxxxx@mail.myuni.edu`, but no with `xxxxxx@myuni.edu` or `xxxxx@mail.example.com`. This should not interfere with manually created users, so when installing Answer or manually creating a user I could do `admin@myuni.edu` and it should work perfectly and allow login.

**Describe alternatives you've considered**
Another option could be allowing to limit emails to match a regex pattern defined by the administrator. This could also affect manually created users because you could just add an or to the regex with the email you want to create the account for. Only side effect of this is that if you manually create many accounts, the regex might get too long and matching the regex with the email would increasingly take longer. On the good side, it is way more flexible than my proposed option.

**Comment by kawa-marcin** [comment]
Until this feature is implemented you can prevent the registration of other domain users by simply adding a constraint on the `user.e_mail`:

So for example, if you want to limit users to be able to register with `<>@answer.dev` email only you could apply:

```sql
ALTER TABLE public.user ADD CONSTRAINT hotjar_email CHECK (e_mail ~* '^[A-Za-z0-9._+%-]+@ansver[.]dev$');
```

> The above query was tested with PosgreSQL

The app will throw an internal DB exception but you prevent unwanted registrations ¯\_(ツ)_/¯

**Comment by fenbox** [comment]
released in [v1.1.0](https://github.com/answerdev/answer/releases/tag/v1.1.0)
